File: example/dart/class/equal.dart
class Newer {
  final x, y;
  Newer(this.x, this.y);

  operator == (N) => this.x == N.x && this.y == N.y;
}

main(List<String> args) {
  var aNewer = new Newer(3, 4);
  var anotherNewer = new Newer(3, 4);
  print(aNewer == anotherNewer);  // true，如果把`operator == (N) ...`这段代码注释则返回false
}

File: example/typescript/class/equal.ts
class Newer {
  readonly _x;
  readonly _y;
  constructor(x, y) {
    this._x = x;
    this._y = y;
  }

  public equals(obj: Newer) {
    return this._x == obj._x && this._y == obj._y
  }
}

const aNewer = new Newer(3, 4)
const anotherNewer = new Newer(3, 4)
console.log(aNewer == anotherNewer)  // false
console.log(aNewer.equals(anotherNewer))  // true